### PR TITLE
Fix error in helioviewer if download dir does not exist

### PR DESCRIPTION
**Pull request by vn-ki**

### CODE REVIEW

File: sunpy/net/tests/test_helioviewer.py [hunk 1]
```diff
@@ -60,3 +60,29 @@ def test_download_jp2(self, client):
...
+    def test_download_jp2_permission_error(self, client):
+        """Tests getJP2Image API method"""
+        with pytest.raises(PermissionError):
+            client.download_jp2(
+                '2020/01/01',
+                observatory='SOHO',
+                instrument='MDI',
+                detector='MDI',
+                measurement='continuum',
+                directory='/')
```
Comment (vn-ki) [inline sunpy/net/tests/test_helioviewer.py]: Would this be a problem on windows machines?

Comment (Cadair) [reply sunpy/net/tests/test_helioviewer.py]: yeah probably. you could use `os.path.sep` but I don't think `\` exists on windows either...

File: sunpy/net/tests/test_helioviewer.py [hunk 2]
```diff
@@ -60,3 +60,29 @@ def test_download_jp2(self, client):
...
+    def test_download_jp2_directory_not_exist(self, client):
+        """Tests getJP2Image API method"""
+
+        filepath = client.download_jp2(
+            '2020/01/01',
+            observatory='SOHO',
+            instrument='MDI',
+            detector='MDI',
+            measurement='continuum',
+            directory='directorynotexist')
```
Comment (Cadair) [inline sunpy/net/tests/test_helioviewer.py]: can you make this inside a tmp dir so it's not related to the path of this code? pytest has a global `tmpdir` fixture, and then this line should be `os.path.join('tmpdir', 'directorynotexist')`

**Comment by Cadair** [comment]
this is missing a changelog

**Pull request merged**